src/components/Modal/BaseModal.js

```diff
@@ -23,166 +24,182 @@ const defaultProps = {
-            shouldAddTopSafeAreaPadding,
-            shouldAddBottomSafeAreaPadding,
-            hideBackdrop,
-        } = getModalStyles(
-            this.props.type,
-            {
-                windowWidth: this.props.windowWidth,
-                windowHeight: this.props.windowHeight,
-                isSmallScreenWidth: this.props.isSmallScreenWidth,
-            },
-            this.props.popoverAnchorPosition,
-            this.props.innerContainerStyle,
-            this.props.outerStyle,
-        );
-        return (
-            <ReactNativeModal
-                onBackdropPress={(e) => {
-                    if (e && e.key === 'Enter') {
-                        return;
-                    }
-                    this.props.onClose();
-                }}
-                // Note: Escape key on web/desktop will trigger onBackButtonPress callback
-                // eslint-disable-next-line react/jsx-props-no-multi-spaces
-                onBackButtonPress={this.props.onClose}
-                onModalShow={() => {
-                    if (this.props.shouldSetModalVisibility) {
-                        Modal.setModalVisibility(true);
-                    }
-                    this.props.onModalShow();
-                }}
-                propagateSwipe={this.props.propagateSwipe}
-                onModalHide={this.hideModal}
-                onSwipeComplete={this.props.onClose}
-                swipeDirection={swipeDirection}
-                isVisible={this.props.isVisible}
-                backdropColor={themeColors.overlay}
-                backdropOpacity={hideBackdrop ? 0 : variables.overlayOpacity}
-                backdropTransitionOutTiming={0}
-                hasBackdrop={this.props.fullscreen}
-                coverScreen={this.props.fullscreen}
-                style={modalStyle}
-                deviceHeight={this.props.windowHeight}
-                deviceWidth={this.props.windowWidth}
-                animationIn={this.props.animationIn || animationIn}
-                animationOut={this.props.animationOut || animationOut}
-                useNativeDriver={this.props.useNativeDriver}
-                hideModalContentWhileAnimating={this.props.hideModalContentWhileAnimating}
-                animationInTiming={this.props.animationInTiming}
-                animationOutTiming={this.props.animationOutTiming}
-                statusBarTranslucent={this.props.statusBarTranslucent}
-                onLayout={this.props.onLayout}
-                avoidKeyboard={this.props.avoidKeyboard}
+        onClose();
+    };
+
+    const {
+        modalStyle,
+        modalContainerStyle,
+        swipeDirection,
+        animationIn: modalStyleAnimationIn,
+        animationOut: modalStyleAnimationOut,
+        shouldAddTopSafeAreaMargin,
+        shouldAddBottomSafeAreaMargin,
+        shouldAddTopSafeAreaPadding,
+        shouldAddBottomSafeAreaPadding,
+        hideBackdrop,
+    } = getModalStyles(
```

Comment:
First I would double check how many times this component gets rerender and if is getting rerendering too many times I would wrap it into a `useMemo`.
`getModalStyles` seems too complex